core/src/test/java/io/methvin/watchservice/DirectoryWatcherTest.java

```diff
@@ -113,6 +112,119 @@ public void validateJdkDirectoryWatcher() throws Exception {
     runWatcher(directory.toPath(), FileSystems.getDefault().newWatchService());
   }
 
+  @Test
+  public void supportFileWatchingOnFiles() throws Exception {
+    File directory = new File(new File("").getAbsolutePath(), "target/directory");
+    FileUtils.deleteDirectory(directory);
+    directory.mkdirs();
+    Path directoryPath = directory.toPath();
+    WatchService watchService = FileSystems.getDefault().newWatchService();
+
+    int waitInMs = 500;
+    FileSystem fileSystem = new FileSystem(directoryPath)
```

Comment:
Let's see if Travis's Xcode instance likes this test, but I don't know why it doesn't seem to pass in my Mac OS High Sierra.